src/components/Authz/Credentials.js

```diff
@@ -83,39 +83,27 @@ export function Credentials({ ...props }) {
 		))
 	}
 
-	return (
-		<>
-			{credentials && (credentials.length !== 0) ?
-				credentials.map((credentialObj, i) => (
-					<div key={i} className='authz-credentials-link' title={credentialObj.username || credentialObj.id}>
-						<i className='bi bi-person pe-1' />
-						<LinkWithAuthz
-							resource={resource}
-							resources={resources}
-							to={`/auth/credentials/${credentialObj.id}`}
-							disabled={hasSeaCatAdminModule}
-						>
-							{credentialObj.username || credentialObj.id}
-						</LinkWithAuthz>
-					</div>
-				))
-			:
-				credentials_ids.map((credentials_id, i) => (
-					<div key={i} className='authz-credentials-link' title={credentials_id}>
-						<i className='bi bi-person pe-1' />
-						<LinkWithAuthz
-							resource={resource}
-							resources={resources}
-							to={`/auth/credentials/${credentials_id}`}
-							disabled={hasSeaCatAdminModule}
-						>
-							{credentials_id}
-						</LinkWithAuthz>
-					</div>
-				))
-			}
-		</>
-	);
+	// Renders a list of credential links
+	const renderList = (list) =>
+		list.map((item, i) => (
+			<div key={i} className='authz-credentials-link' title={item.username || item.id}>
+				<i className='bi bi-person pe-1' />
+				<LinkWithAuthz
+					resource={resource}
+					resources={resources}
+					to={`/auth/credentials/${item.id}`}
+					disabled={hasSeaCatAdminModule}
+				>
+					{item.username || item.id}
+				</LinkWithAuthz>
+			</div>
+		));
+
+	// If credentials are available, render them; otherwise, use IDs as placeholders
+	const content = (credentials.length !== 0) ? renderList(credentials) : renderList(credentials_ids.map(id => ({ id, username: id })));
```

Comment:
@aringocode Why cant it be:

```
const content = (credentials.length !== 0) ? renderList(credentials) : renderList(credentials_ids);
```
? Why is it necessary to map the credentials_ids when the maping logic is the same for `credentials` and for `credentials_ids` ?